vulcan/lib/server/workflows/scripts/plot_umap.py

```diff
@@ -55,29 +55,32 @@ def get(ids, value):
     color = None
 
 ##### OUTPUT
+plot_args = {
+    'data_frame': scdata.obsm['X_umap'],
+    'x':0, 'y':1,
+    'color_discrete_sequence': colors,
+    'color': color
+    }
+
+# if 'color is dicrete'
+plot_args['category_orders'] = {'color': order(unique(color)) }
+
 if custom_tooltip:
     dat = pd.DataFrame(scdata.obsm['X_umap'])
     dat[hover_name] = hover_text
-    fig = px.scatter(
-        dat, x=0, y=1,
-        color_discrete_sequence=colors,
-        color=color,
-        hover_data=[hover_name]
-        )
-else:
-    fig = px.scatter(
-        scdata.obsm['X_umap'], x=0, y=1,
-        color_discrete_sequence=colors,
-        color=color)
+    plot_args['data_frame'] = dat
+    plot_args['hover_data'] = [hover_name]
+
+fig = do_call(px.scatter, plot_args)
```

Comment:
rather than relying on this, do_call function, why not inline just with `px.scatter(**plot_args)` ?
The do_call function seems superfulous.

Reply:
No reason other than that I'm a python noob.  I'll make it that instead.